feat(function): Add varbinary variants for strpos and contains (#15809)

laurencer opened the pull request:
Summary:

Adds `strpos`, `strrpos` and `contains` for varbinary columns which will allow for efficient string searching in encoded text data where we don't want to pay the charset decoding price (i.e. we don't have to convert the column to utf-8 and just do byte matches as an approximation).

Reviewed By: kaikalur

Differential Revision: D89479354

zacw7 reviewed (changes requested):
could you pls add the doc here: https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/presto/binary.rst

laurencer commented:
> could you pls add the doc here: https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/presto/binary.rst

@zacw7 added docs that match the string and array variants

zacw7 reviewed (approved):
LGTM. Thanks for adding the functions.

laurencer committed:
feat(function): Add varbinary variants for strpos and contains (#15809)

Summary:
Pull Request resolved: https://github.com/facebookincubator/velox/pull/15809

Adds `strpos`, `strrpos` and `contains` for varbinary columns which will allow for efficient string searching in encoded text data where we don't want to pay the charset decoding price (i.e. we don't have to convert the column to utf-8 and just do byte matches as an approximation).

Reviewed By: kaikalur

Differential Revision: D89479354
